client/selfrepair/tests/test_normandy_driver.js

```diff
@@ -36,21 +39,106 @@ describe('Normandy Driver', () => {
     });
   });
 
+  describe('saveHeartbeatFlow', () => {
+    const driver = new NormandyDriver();
+
+    afterEach(() => {
+      fetchMock.restore();
+    });
+
+    beforeEach(() => {
+      fetchMock.get(urlPathMatcher('/api/v2/hb/'), 200);
+    });
+
+    it('should not POST data if in testing mode', async () => {
+      driver.testing = true;
+      await driver.saveHeartbeatFlow({
+        flowId: '123456789',
+      });
+
+      expect(fetchMock.calls().matched.length).toEqual(0);
+    });
+
+    it('should POST flow data to input', async () => {
+      driver.testing = false;
+      await driver.saveHeartbeatFlow({
+        flowId: '123456789',
+      });
+
+      expect(fetchMock.lastOptions()).toEqual({
+        method: 'POST',
+        body: '{"flowId":"123456789"}',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      });
+    });
+  });
+
   describe('client', () => {
-    it("should fetch the user's distribution ID from UITour", async () => {
+    it("should fetch the user's config from UITour", async () => {
       const uitour = {
         getConfiguration(config, cb) {
-          if (config === 'appinfo') {
-            cb({ distribution: 'funnelcake85' });
-          } else {
-            cb({});
+          switch (config) {
+            case 'sync':
+              return cb({
+                setup: false,
+              });
+            case 'appinfo':
+              return cb({
+                version: '50.0.2',
+                defaultUpdateChannel: 'aurora',
+                defaultBrowser: true,
+                distribution: 'funnelcake85',
+              });
+            case 'selectedSearchEngine':
+              return cb({ searchEngineIdentifier: 'Yahoo' });
+            default:
+              return cb({});
           }
         },
       };
       const driver = new NormandyDriver(uitour);
-
       const client = await driver.client();
+
+      expect(client.syncSetup).toEqual(false);
       expect(client.distribution).toEqual('funnelcake85');
+      expect(client.isDefaultBrowser).toEqual(true);
+      expect(client.searchEngine).toEqual('Yahoo');
+    });
+  });
+
+  describe('uuid', () => {
+    it('returns a valid userID', () => {
+      const UUID_ISH_REGEX = /^[a-f0-9-]{36}$/;
+      const driver = new NormandyDriver();
+      const uuid = driver.uuid();
+      expect(UUID_ISH_REGEX.test(uuid)).toBe(true);
+    });
+  });
+
+  describe('log', () => {
+    const driver = new NormandyDriver();
+
+    it('will not log debug level unless in testing mode', () => {
+      spyOn(console, 'log');
+      driver.log('lorem ipsum');
```

Comment:
Should probably set `driver.testing = false` here to be explicit (also, to match the "should POST flow data to input" test).